WordPressAuthenticatorTests/Authenticator/WordPressAuthenticatorTests.swift

```diff
@@ -38,21 +45,352 @@ class WordPressAuthenticatorTests: XCTestCase {
+    }
+    
+    func testSignInForWPOrgReturnsVC() {
+        let vc = WordPressAuthenticator.signinForWPOrg()
+        
+        XCTAssertTrue((vc as Any) is LoginSiteAddressViewController)
+    }
+    
+    func testShowLoginFromPresenterReturnsLoginInitialVC() {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+        
+        WordPressAuthenticator.showLoginFromPresenter(presenterSpy, animated: true)
+        wait(for: [expectation], timeout: 3)
+        
+        XCTAssertTrue(presenterSpy.presentedVC is LoginNavigationController)
+    }
+    
+    func testShowLoginForJustWPComPresentsCorrectVC() {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+        
+        WordPressAuthenticator.showLoginForJustWPCom(from: presenterSpy)
+        wait(for: [expectation], timeout: 3)
+        
+        XCTAssertTrue(presenterSpy.presentedVC is LoginNavigationController)
+    }
+    
+    func testShowLoginForJustWPComTracksOpenedLogin() {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let delegateSpy = WordPressAuthenticatorDelegateSpy()
+        WordPressAuthenticator.shared.delegate = delegateSpy
+        
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+        
+        WordPressAuthenticator.showLoginForJustWPCom(from: presenterSpy)
+        wait(for: [expectation], timeout: 3)
+        
+        let trackedEvent = delegateSpy.trackedElement
+        
+        XCTAssertEqual(trackedEvent, WPAnalyticsStat.openedLogin)
+    }
+    
+    func testShowLoginForJustWPComSetsMetaProperties() {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+        
+        WordPressAuthenticator.showLoginForJustWPCom(from: presenterSpy,
+                                                     xmlrpc: "https://example.com/xmlrpc.php",
+                                                     username: "username",
+                                                     connectedEmail: "email-address@example.com")
+        
+        guard let navController = presenterSpy.presentedVC as? LoginNavigationController, let controller = navController.viewControllers.first as? LoginEmailViewController else {
+            XCTFail("Could not fetch correct ViewController")
+            return
+        }
```

Comment:
XCTest has an API made to simplify writing this kind of checks: [`XCTUnwrap`](https://developer.apple.com/documentation/xctest/3380195-xctunwrap).